### PR TITLE
TEST-#4237: parallelize Dask tests on windows

### LINKED ISSUE #4237: parallelize Dask tests on windows

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4237 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [ ] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -575,11 +575,15 @@ jobs:
         run: |
           conda info
           conda list
+      - run: python -m pytest -n 2 ${{matrix.test-task}}
+        if: ${{ (matrix.engine == 'dask') && (matrix.test-task != 'modin/pandas/test/test_io.py') && (matrix.test-task != 'modin/pandas/test/test_concat.py') }}
```
Comment (anmyachev) [inline .github/workflows/ci.yml:579]: `test_concat.py` file also does not parallelize in our tests on ubuntu.

**Comment by anmyachev** [comment]
The slowest test - `test_join_sort.py` now runs more than twice as fast.

**Comment by anmyachev** [comment]
@gshimansky can you review this?

**Review by devin-petersohn — COMMENTED** [review]
These tests are very short and the overhead of doing `xdist` might be higher than the tests take to run. Do we need to do this?

**Comment by anmyachev** [comment]
> These tests are very short and the overhead of doing `xdist` might be higher than the tests take to run. Do we need to do this?

I did by analogy with the tests in `test-all` job, but there is no acceleration.

**Pull request closed without merging**